File: 2.1 Algorithmic Toolbox/Week 3 Greedy Algorithms/02_greedy_algorithms_starter_files/change/change.py
# Uses python3
# import sys

def get_change(m):
    #write your code here
    coins = [10, 5, 1]
    num = [0, 0, 0]
    cn = len(coins)
    remainder = m
    for i in range(cn):
        num[i] = int(remainder / coins[i])
        remainder = int(remainder % coins[i])
    return sum(num)

if __name__ == '__main__':
    m = int(input())
    print(get_change(m))

File: 2.1 Algorithmic Toolbox/Week 3 Greedy Algorithms/02_greedy_algorithms_starter_files/change/change.cpp
#include <iostream>

int get_change(int m) {
	int coins[3] = {10, 5, 1};
	int result = 0;
	int remainder = m;
	for (int i = 0; i < 3; i++){
		result += remainder / coins[i];
		remainder = remainder % coins[i];
	}
  return result;
}

int main() {
  int m;
  std::cin >> m;
  std::cout << get_change(m) << '\n';
  system("pause");
}
